DuckDuckGo/AppDependencyProvider.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  AppDependencyProvider.swift
+//  DuckDuckGo
+//
+//  Copyright © 2018 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Core
+
+protocol DependencyProvider {
+    var appSettings: AppSettings { get }
+    var variantManager: VariantManager { get }
+}
+
+/// Provides dependencies for objects that are not directly instantiated
+/// through `init` call (e.g. ViewControllers created from Storyboards).
+class AppDependencyProvider: DependencyProvider {
```

Comment:
Yes, naming things :)

`ViewControllersDependencyProvider` is good, tho:
`class AppViewControllersDependencyProvider: ViewControllersDependencyProvider` 
is a bit long :)

Alternative 1:
`class AppDependencyProvider: ViewControllersDependencyProvider`
... to keep it shorter, or...

Alternative 2:
`class AppMVCDependencyProvider: MVCDependencyProvider`
... to make it more related to MVC, so in theory it will be obsolete once we replace MVC with something else?

Reply:
I'm starting to think this is a distraction, sorry. What is there works - we'll just have to watch out in reviews to make sure that if we use the `AppDependencyProvider` we're doing it for the right reason.

I look forward to our future architecture discussion!  :)